### PR TITLE
Refactor and add socket activity check for internet

### PR DESCRIPTION
**Pull request by fhunleth**
This commit series deals with improving internet connectivity checking. This is
not the last PR on this topic, but it's an improvement to where VintageNet has
been on this topic.

Internet connectivity checking is important for multi-homed devices so that the
routing tables can be set up to prefer faster and cheaper interfaces. When those
interfaces lose internet connectivity, the routing tables should be updated so
reconnects go to the new best interface.

The first two commits move code around so that it's easier to improve internet
connectivity checking. All of the feedback on connectivity checking has been
nontrivial enough that mixing it in with the general network interface code is
just too much. The "ping" code is now called `TCPPing` to accurately reflect
what it does. Moving the "3-strikes" ping failure state machine to its own
module was also needed since it was complicated enough that I was worried about
messing it up accidentally. Now the state machine is completely pure code and
easier to test.

The third and fourth commits add a socket inspector. The socket inspector looks
at TCP connections going through the network interface of interest and going
somewhere off-LAN. If the tx and rx counters on those sockets both go up between
checks, then there was internet connectivity over the past sampling period. This
means that a ping of an external host can be skipped. The keepalive messages on
MQTT and websocket connections will satisfy this check. This has two advantages:
less bandwidth usage and it works on tightly firewalled installs.

The final commit makes the internet connectivity checker try the socket
inspector first and if that fails, it tries to connect to an external host like
before.

These updates are backwards compatible, but they'll log a warning to update code
that refers to `VintageNet.Interface.InternetConnectivityChecker` to
`VintageNet.Connectivity.InternetChecker`.

- Move connectivity check code to its own folder
- Move check handling logic into its own module
- Add internet connectivity inspector
- inspector: support the socket API too
- Use the connectivity inspector as a first internet check

### CODE REVIEW

**Comment by fhunleth** [comment]
FYI - I need to fix some warnings and test failures. Hopefully we'll have green lights soon. I've been verifying this on test devices. It's a big update, so depending on the comments, I'm ready to break it up into smaller chunks to help reviews if needed.

File: lib/vintage_net/connectivity/inspector.ex [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  1. No data is sent to perform the check which is good for metered connections
+  2. Most long-lived TCP connections have a keepalive mechanism that generates
+     traffic, so this piggy-backs off that existing connectivity check.
+  3. Devices can be behind very restrictive firewalls and internet connectivity
+     can still be verified without knowing which IP/port/protocol combinations
+     are allowed.
+
+  It is not perfect:
+
+  1. It only works on long-lived connections TCP connections.
```
Comment (jjcarstens) [inline lib/vintage_net/connectivity/inspector.ex:22]:
```suggestion
  1. It only works on long-lived TCP connections.
```

**Review by jjcarstens — APPROVED** [review]
LGTM

**Review by mattludwigs — APPROVED** [review]

**Comment by fhunleth** [comment]
@jjcarstens confirmed that this makes a very tightly firewalled device report internet connectivity now.

**Pull request merged**